### PR TITLE
Fix the sporadic issue about the IEClassBasicTest.smoke_LoadNetworkToDefaultDeviceNoThrow failure

### PR DESCRIPTION
**Pull request by yangwang201911**
In order to avoid this sporadic crash issue,  adding the logic of loading network to CPU device only when running CPU functional test on CI.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

### Details:
 - Update the test cases for both the LoadNetwork and compile_model API

### Tickets:
 - https://jira.devtools.intel.com/browse/CVS-75022

### CODE REVIEW

**Review by peterchen-intel — APPROVED** [review]

**Review by wangleis — APPROVED** [review]
default device test case on CPU will be covered in another PR

File: src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_executable_network/exec_network_base.cpp [hunk 1]
```diff
@@ -51,9 +51,7 @@ namespace {
 
     const std::vector<std::map<std::string, std::string>> configSetPrc = {
             {},
-            {{InferenceEngine::PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, InferenceEngine::PluginConfigParams::CPU_THROUGHPUT_AUTO}}
-    };
-
+            {{InferenceEngine::PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, InferenceEngine::PluginConfigParams::CPU_THROUGHPUT_AUTO}} };
```
Comment (alalek) [inline src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_executable_network/exec_network_base.cpp:54]: We don't really need this change.

Comment (yangwang201911) [reply src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_executable_network/exec_network_base.cpp]: Updated.

**Review by myshevts — APPROVED** [review]

**Comment by myshevts** [comment]
from what I see this is really moving the instantiations to the invidiual plugins, rather than transparent instatitions everywhere (via the shared tests). This should guarantee that the particular plugin (e.g. CPU or GPU) is operable, not just compilable (as presumably the sporadic  failures originate in the cases when e.g. the GPU driver is not installed correctly, so GPU may crash on such CI machines)

File: src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_network_base.hpp [hunk 2]
```diff
@@ -77,12 +77,6 @@ TEST_P(OVExecutableNetworkBaseTest, canLoadCorrectNetworkToGetExecutable) {
     EXPECT_NO_THROW(auto execNet = core->compile_model(function, targetDevice, configuration));
 }
 
-TEST(OVExecutableNetworkBaseTest, smoke_LoadNetworkToDefaultDeviceNoThrow) {
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_network_base.hpp:80]: Why did you remove these tests from shared lib? They should be here

Comment (yangwang201911) [reply src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_network_base.hpp:80]: This shared test case in the CPU functional test will load GPU plugin and cause the sporadic crash issue. So we the move this test case to individual plugins(CPU, GPU and Myriad plugin) to fix this issue.

Comment (iefode) [reply src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_network_base.hpp:80]: we must use `targetDevice` in this case (`EXPECT_NO_THROW(auto execNet = core->compile_model(function, targetDevice));` and `ASSERT_NO_THROW(ie.LoadNetwork(actualCnnNetwork, targetDevice));`). Please, return these case to the lib using the proposed changes and I will merge it. Cases without using of device name should be added to cpu functional tests (the same cases should be removed from other plugin functional tests).

Comment (yangwang201911) [reply src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_network_base.hpp:80]: Do you mean we should use `targetDevice ` in the removed test case for both `compiled_model` and `LoadNetwork` and keep those test cases in the shared lib?  We added the overloading version of those  2 APIs to support default device name(AUTO) and also enabled the corresponding test cases. More detailed information you can refer to #8956 . These test cases aims at validating the `compiled_model` and `LoadNetwork`  API using default device instead of receiving the device name from the user. Then we found these 2 test cases added in #8956 causes sporadic crash issue. So we moved these 2 test cases from the shared lib to individual plugins to fix this sporadic crash issue in this PR.

Comment (peterchen-intel) [reply src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_network_base.hpp:80]: Hi Irina, this case was added to test LoadNetwork and compile_model without deviceName/device_name only.  They are new APIs to enable user call without device name.  Removal of this test case doesn't impact others. https://github.com/openvinotoolkit/openvino/pull/8956. As default implementation LoadNetwork will use AUTO plugin if deviceName is omitted, AUTO plugin will try to load to GPU when it is available. Unfortunately, the test cases in CPU functional tests will be run on selected nodes which guarantee only CPU works well. This why the issue https://jira.devtools.intel.com/browse/CVS-75022 happens (the selected node works OK for CPU, but works bad for GPU even it is available).

Comment (iefode) [reply src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_network_base.hpp:80]: We found the commit with the tests adding. I have reviewd it and left comment. I disagree with the implementation. https://github.com/openvinotoolkit/openvino/pull/8956/files

**Review by iefode — CHANGES_REQUESTED** [review]

**Comment by peterchen-intel** [comment]
Replaced with https://github.com/openvinotoolkit/openvino/pull/9617
1. Add macro SKIP_IF_CURRENT_TEST_IS_DISABLED() 
2. Disable it in CPUFunTests
